modules/files/files_test.go

```diff
@@ -48,6 +63,27 @@ func TestIsExistingDir(t *testing.T) {
 	assert.True(t, IsExistingDir(currentFileDir))
 }
 
+func TestCopyFolderToTemp(t *testing.T) {
+	t.Parallel()
+
+	expectedErrorType := new(os.PathError)
+	tempFolderPrefix := "someprefix"
+	currentFile, err := filepath.Abs(os.Args[0])
+	require.NoError(t, err)
+
+	currentFileDir := filepath.Dir(currentFile)
+	filter := func(path string) bool {
+		return !PathContainsHiddenFileOrFolder(path) && !PathContainsTerraformState(path)
+	}
+
+	_, err = CopyFolderToTemp("/not/a/real/path", tempFolderPrefix, filter)
+	assert.IsType(t, expectedErrorType, err)
+
+	folder, err := CopyFolderToTemp(currentFileDir, tempFolderPrefix, filter)
```

Comment:
We shouldn't copy anything to the current working dir, as that will leave the file system in a messy state. You can use `ioutil.TempDir` to create a dir in the system temp folder. 

Also, the checks in this test don't seem to work. One check is that `CopyFolderToTemp` handles in valid paths correctly, which is fine. But shouldn't the other check, the one related to this PR, be that it _doesn't_ overwrite an existing folder?

Reply:
Update to use a temp dir and also fixed the checks to cover what you described, let me know if those make more sense now!